tw-base-utils/src/main/java/com/transferwise/common/baseutils/concurrency/DefaultExecutorServicesProvider.java

```diff
@@ -2,27 +2,34 @@
 
 import java.util.concurrent.ExecutorService;
 import java.util.concurrent.Executors;
-import javax.annotation.PostConstruct;
-import javax.annotation.PreDestroy;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+import org.springframework.beans.factory.DisposableBean;
+import org.springframework.beans.factory.InitializingBean;
 
-public class DefaultExecutorServicesProvider implements IExecutorServicesProvider {
+public class DefaultExecutorServicesProvider implements IExecutorServicesProvider, InitializingBean, DisposableBean {
 
   private ExecutorService executorService;
   private ScheduledTaskExecutor scheduledTaskExecutor;
 
   private boolean initialized = false;
 
-  @PostConstruct
-  public synchronized void init() {
-    if (!initialized) {
-      executorService = Executors.newCachedThreadPool(new CountingThreadFactory("tw-base"));
-      scheduledTaskExecutor = new SimpleScheduledTaskExecutor(null, executorService);
-      scheduledTaskExecutor.start();
-      initialized = true;
+  private Lock initializationLock = new ReentrantLock();
+
+  public void afterPropertiesSet() {
```

Comment:
Just getting rid of "synchronized" as it is not working the best with virtual threads.